src/clients/api/mutations/useSupply.ts

```diff
@@ -0,0 +1,65 @@
+import { useMutation, MutationObserverOptions } from 'react-query';
+
+import { VTokenId } from 'types';
+import queryClient from 'clients/api/queryClient';
+import supplyBnb, { ISupplyBnbInput, SupplyBnbOutput } from 'clients/api/mutations/supplyBnb';
+import supply, { ISupplyInput, SupplyOutput } from 'clients/api/mutations/supply';
+import FunctionKey from 'constants/functionKey';
+import { useVTokenContract } from 'clients/contracts/hooks';
+import { getVTokenContract } from 'clients/contracts';
+
+interface IUseSupplyArgs {
+  assetId: VTokenId;
+  account: string | undefined;
+}
+
+export type UseSupplyParams =
+  | Omit<ISupplyInput, 'tokenContract' | 'assetId' | 'account'>
+  | Omit<ISupplyBnbInput, 'tokenContract' | 'assetId' | 'account'>;
+
+const getWeb3 = (params: UseSupplyParams) => {
+  if ('web3' in params) {
+    return params.web3;
+  }
+  return undefined;
+};
+
+const useSupply = (
+  { assetId, account }: IUseSupplyArgs,
+  // TODO: use custom error type https://app.clickup.com/t/2rvwhnt
+  options?: MutationObserverOptions<SupplyOutput | SupplyBnbOutput, Error, UseSupplyParams>,
+) => {
+  const vContract = useVTokenContract(assetId as VTokenId);
+  return useMutation(
+    [FunctionKey.SUPPLY, assetId],
+    params => {
+      const web3 = getWeb3(params);
+      const { amount } = params;
+      if (assetId === 'bnb' && web3) {
+        const vBnbContract = getVTokenContract('bnb', web3);
+        supplyBnb({
+          tokenContract: vBnbContract,
+          web3,
+          account,
+          amount,
+        });
+      }
+      return supply({
+        tokenContract: vContract,
+        account,
+        amount,
+      });
+    },
```

Comment:
I feel like this should be two separate hooks, for these reasons:
- `vContract` is always declared, but unused if `assetId` is `bnb` or if no `web3` instance was provided
- `supplyBnb` and `supply` have expect different inputs: `supplyBnb` expects `web3` to be provided while `supply` doesn't

Separating these into two different hooks would solve these issues, and also means you could write `useSupplyBnb` so it fetched the `web3` instance automatically using the `useWeb3` hook. This way its `mutate` function would only need to be passed the account address and an amount.